tests/src/JIT/HardwareIntrinsics/X86/Shared/GenerateTests.csx

```diff
@@ -162,6 +162,9 @@ private static readonly (string templateFileName, Dictionary<string, string> tem
     ("SimpleBinOpTest.template",    new Dictionary<string, string> { ["Isa"] = "Sse2", ["LoadIsa"] = "Sse2", ["Method"] = "Or",                           ["RetVectorType"] = "Vector128", ["RetBaseType"] = "UInt32", ["Op1VectorType"] = "Vector128", ["Op1BaseType"] = "UInt32", ["Op2VectorType"] = "Vector128", ["Op2BaseType"] = "UInt32",                                          ["LargestVectorSize"] = "16", ["NextValueOp1"] = "TestLibrary.Generator.GetUInt32()", ["NextValueOp2"] = "TestLibrary.Generator.GetUInt32()", ["ValidateFirstResult"] = "(uint)(left[0] | right[0]) != result[0]",                                                                                            ["ValidateRemainingResults"] = "(uint)(left[i] | right[i]) != result[i]"}),
     ("SimpleBinOpTest.template",    new Dictionary<string, string> { ["Isa"] = "Sse2", ["LoadIsa"] = "Sse2", ["Method"] = "Or",                           ["RetVectorType"] = "Vector128", ["RetBaseType"] = "UInt64", ["Op1VectorType"] = "Vector128", ["Op1BaseType"] = "UInt64", ["Op2VectorType"] = "Vector128", ["Op2BaseType"] = "UInt64",                                          ["LargestVectorSize"] = "16", ["NextValueOp1"] = "TestLibrary.Generator.GetUInt64()", ["NextValueOp2"] = "TestLibrary.Generator.GetUInt64()", ["ValidateFirstResult"] = "(ulong)(left[0] | right[0]) != result[0]",                                                                                           ["ValidateRemainingResults"] = "(ulong)(left[i] | right[i]) != result[i]"}),
+    ("SimpleBinOpTest.template",    new Dictionary<string, string> { ["Isa"] = "Sse2", ["LoadIsa"] = "Sse2", ["Method"] = "PackSignedSaturate",           ["RetVectorType"] = "Vector128", ["RetBaseType"] = "Int16",  ["Op1VectorType"] = "Vector128", ["Op1BaseType"] = "Int32",  ["Op2VectorType"] = "Vector128", ["Op2BaseType"] = "Int32",                                           ["LargestVectorSize"] = "16", ["NextValueOp1"] = "TestLibrary.Generator.GetInt32()",  ["NextValueOp2"] = "TestLibrary.Generator.GetInt32()",  ["ValidateFirstResult"] = "Sse2Verify.PackSignedSaturate(result, left, right)",                                                                                 ["ValidateRemainingResults"] = "Sse2Verify.PackSignedSaturate(result, left, right)"}),
```

Comment:
nit: We haven't been doing `Sse2Verify.PackSignedSaturate` anywhere else and have been inlining the validation logic instead.

Reply:
After trying doing this for packing shorts with 16 `sbyte` results inlining got really messy, see (code does not contain saturating conversion which is in another place):

https://github.com/dotnet/coreclr/blob/678243d625e7f5d537c42095dea8a5ef10142293/tests/src/JIT/HardwareIntrinsics/X86/Sse2/Sse2Verify.cs#L81-L132

Reply:
Just one more point - large groups of intrinsics tests execute quite slow. IMHO we could speed them up by resigning from loops during checks (in particular where there are many members to compare) and instead do a single check on whole buffer / `VectorNNN`. This would require some template rewriting but should be easy to do.

Reply:
> After trying doing this for packing shorts with 16 sbyte results inlining got really messy, see (code does not contain saturating conversion which is in another place)

Why are you trying to extract each byte and compare individually, rather than codifying the logic for selecting left vs right:
Ex: https://github.com/dotnet/coreclr/blob/678243d625e7f5d537c42095dea8a5ef10142293/tests/src/JIT/HardwareIntrinsics/X86/Shared/GenerateTests.csx#L355

Reply:
AFAIR when we discussed the way tests will validate result we agreed to do it in clear way to show how given intrinsic works. The reason why I would prefer not to use:
```C#
result[i2] != ((right[i3 - inner] > 0xFFFF) ? 0xFFFF : ((right[i3 - inner] < 0) ? 0 : BitConverter.ToUInt16(BitConverter.GetBytes(right[i3 - inner]), 0)))
``` 

as it does not tell very much about functionality.

Reply:
There were many discussions about the tests, and all I remember is that we are not testing the implementation, but that the right instruction is emitted, and in order to validate that (given that their isn't a good way to check that "Foo()" directly emits "ins"), we will do a minimal functionality test.

I don't remember anything being decided about making the tests "clear" on how the intrinsic works (although my memory could just be faulty).

In any case, the current tests are fairly standardized on the existing mechanism, and the inline ternary code isn't overly complicated to understand (or to compare to the diagrams/descriptions in the architecture manuals)

Reply:
I am, however, fine with moving to the new model if we can all reach a consensus that it is a better approach.

@fiigii, @CarolEidt, @eerhardt. Thoughts?

Reply:
> we are not testing the implementation, but that the right instruction is emitted, and in order to validate that (given that their isn't a good way to check that "Foo()" directly emits "ins"), we will do a minimal functionality test.

Yes, that's my understanding as well. We are not undertaking an effort to validate the hardware. That said, we need to have sufficient functionality testing to be able to use stress testing to identify encoding bugs that slip by when we happily encode an [x|y]mm0 where it doesn't belong.